api/src/Entity/ActivityProgressLabel.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
+use ApiPlatform\Metadata\ApiFilter;
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Delete;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Patch;
+use ApiPlatform\Metadata\Post;
+use App\InputFilter;
+use App\Repository\ActivityProgressLabelRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * Progress labels in a camp.
+ * To each activity one label can be assigned.
+ */
+#[ApiResource(
+    operations: [
+        new Get(
+            normalizationContext: self::ITEM_NORMALIZATION_CONTEXT,
+            security: 'is_granted("CAMP_COLLABORATOR", object) or is_granted("CAMP_IS_PROTOTYPE", object)'
+        ),
+        new Patch(
+            normalizationContext: self::ITEM_NORMALIZATION_CONTEXT,
+            security: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)',
+            validationContext: ['groups' => ['Default', 'update']]
+        ),
+        new Delete(
+            security: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)'
+        ),
+        new GetCollection(
+            security: 'is_authenticated()'
+        ),
+        new Post(
+            validationContext: ['groups' => ['Default', 'create']],
+            denormalizationContext: ['groups' => ['write', 'create']],
+            normalizationContext: self::ITEM_NORMALIZATION_CONTEXT,
+            securityPostDenormalize: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)'
+        ),
+    ],
+    denormalizationContext: ['groups' => ['write']],
+    normalizationContext: ['groups' => ['read']],
+    order: ['camp.id', 'position']
+)]
+#[ApiFilter(filterClass: SearchFilter::class, properties: ['camp'])]
+#[ORM\Entity(repositoryClass: ActivityProgressLabelRepository::class)]
+// TODO: Gedmo + UniqueConstraint
+// #[ORM\UniqueConstraint(name: 'camp_activity_progress_label', columns: ['campid', 'position'])]
```

Comment:
The only problem with the unique constraint was that i did not rename the name here too.
https://github.com/BacLuc/ecamp3/actions/runs/4831483898/jobs/8608997636
(Sorry for that)

Either remove the unique constraint comment (Because Gedmo/Sortable also assures uniqueness) or add it

Reply:
as far as I can see, the problem is that Gedmo can't handle the unique constraint (which wouldn't be easy either - but nice).

So I suspect that we will have to go without it. Unfortunately

Reply:
In my case it worked when the constraint is `DEFERRABLE INITIALLY DEFERRED`, but without it is also ok

Reply:
oh - cool - let me try